src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+            }
+
+            void symbolValidator(ModuleSymbol m)
+            {
+                bool isSource = !(m is PEModuleSymbol);
+                var c = (NamedTypeSymbol)m.GlobalNamespace.GetMember("C");
+
+                var property = (PropertySymbol)c.GetMembers("Property").Single();
+                // PROTOTYPE(records): adjust SymbolDisplayVisitor once we have a public IsInitOnly API
+                Assert.Equal("System.String C.Property { get; set; }", property.ToTestDisplayString());
+                var propertyAttributes = property.GetAttributes().Select(a => a.ToString());
+                AssertEx.Empty(propertyAttributes);
+
+                var getterAttributes = property.GetMethod.GetAttributes().Select(a => a.ToString());
+                if (isSource)
+                {
+                    AssertEx.Empty(getterAttributes);
+                }
+                else
+                {
+                    AssertEx.Equal(new[] { "System.Runtime.CompilerServices.CompilerGeneratedAttribute" }, getterAttributes);
+                }
+
+                var setterAttributes = property.SetMethod.GetAttributes().Select(a => a.ToString());
+                var modifier = property.SetMethod.ReturnTypeWithAnnotations.CustomModifiers.Single();
+                Assert.Equal("System.Runtime.CompilerServices.IsExternalInit", modifier.Modifier.ToTestDisplayString());
+                Assert.False(modifier.IsOptional);
+
+                if (isSource)
+                {
+                    AssertEx.Empty(setterAttributes);
+                }
+                else
+                {
+                    AssertEx.Equal(new[] { "System.Runtime.CompilerServices.CompilerGeneratedAttribute" }, setterAttributes);
+                }
+
+                var backingField = (FieldSymbol)c.GetMembers("<Property>k__BackingField").Single();
+                var backingFieldAttributes = backingField.GetAttributes().Select(a => a.ToString());
+                Assert.True(backingField.IsReadOnly);
+                if (isSource)
+                {
+                    AssertEx.Empty(backingFieldAttributes);
+                }
+                else
+                {
+                    AssertEx.Equal(
+                        new[] { "System.Runtime.CompilerServices.CompilerGeneratedAttribute",
+                            "System.Diagnostics.DebuggerBrowsableAttribute(System.Diagnostics.DebuggerBrowsableState.Never)" },
+                        backingFieldAttributes);
+
+                    var peBackingField = (PEFieldSymbol)backingField;
+                    Assert.Equal(System.Reflection.FieldAttributes.InitOnly | System.Reflection.FieldAttributes.Private, peBackingField.Flags);
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void AssignmentDisallowed_PE(bool emitImage)
```

Comment:
>_PE [](start = 40, length = 3)

What is the "_PE" suffix is supposed to convey? It doesn't look like the scenario is always PE specific

Reply:
This scenario explicitly tests PE symbols, while other "AssignmentDisallowed" tests don't. This allows to find the test

---
In reply to: [415203299](https://github.com/dotnet/roslyn/pull/43275#discussion_r415203299) [](ancestors = 415203299)